gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/user/model/BaseUserEntity.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.user.model;
+
+import java.util.Date;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class BaseUserEntity {
+
+    private String id;
+    private String organizationId;
+    private String firstname;
+    private String lastname;
+    private String email;
+    private Date createdAt;
+    private Date updatedAt;
+    /** The source when user is coming from an external system (LDAP, ...) */
+    private String source;
+    /** The user reference in the external source */
+    private String sourceId;
+
+    public String displayName() {
+        if ((firstname != null && !firstname.isEmpty()) || (lastname != null && !lastname.isEmpty())) {
+            if (firstname != null && !firstname.isEmpty()) {
+                return firstname + ((lastname != null && !lastname.isEmpty()) ? ' ' + lastname : "");
+            }
+            return lastname;
+        } else if (email != null && !email.isEmpty() && !"memory".equals(source)) {
+            return email;
+        }
```

Comment:
Don't know if it's clearer like that but it would do the same no ?

```suggestion
        if (StringUtils.isNotBlank(firstname)) {
            if (StringUtils.isNotBlank(lastname)) {
                return firstname + ' ' + lastname;
            }
            return firstname;
        } else if (StringUtils.isNotBlank(lastname)) {
            return lastname;
        } else if (StringUtils.isNotBlank(email) && !"memory".equals(source)) {
            return email;
        }
```